h2o-core/src/main/java/water/api/schemas3/ModelParametersSchemaV3.java

```diff
@@ -162,7 +162,7 @@
    * Metric to use for convergence checking, only for _stopping_rounds > 0
    */
 //  @API(help = "Metric to use for early stopping (AUTO: logloss for classification, deviance for regression)", values = {"AUTO", "deviance", "logloss", "MSE", "RMSE","MAE","RMSLE", "AUC", "lift_top_group", "misclassification", "mean_per_class_error", "r2"}, level = API.Level.secondary, direction=API.Direction.INOUT, gridable = true)
-  @API(help = "Metric to use for early stopping (AUTO: logloss for classification, deviance for regression)", values = {"AUTO", "deviance", "logloss", "MSE", "RMSE","MAE","RMSLE", "AUC", "lift_top_group", "misclassification", "mean_per_class_error"}, level = API.Level.secondary, direction=API.Direction.INOUT, gridable = true)
+  @API(help = "Metric to use for early stopping (AUTO: logloss for classification, deviance for regression)", values = {"AUTO", "deviance", "logloss", "MSE", "RMSE","MAE","RMSLE", "AUC", "lift_top_group", "misclassification", "mean_per_class_error", "custom", "custom_increasing"}, level = API.Level.secondary, direction=API.Direction.INOUT, gridable = true)
```

Comment:
@michalkurka, FYI I created an EnumValuesProvider to avoid having to sync those enums.
see [usage example](https://github.com/h2oai/h2o-3/blob/29cb7b4ecbcf5b0714ea72732f7b048c29541e58/h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java#L126)

Reply:
That looks nice, it is useful in those case when you want to expose all the values in the Enum (=> none of them are internal). Thank you

Reply:
@seb-h2o I tried, it doesn't work for ModelBuilders (values annotation is expected). Needs a modification. I will submit a PR. But do we really need the Provider classes? Wouldn't it be enough to just have a static method that converts enum to list of string values? like this:

    public static <E extends Enum<E>> String[] getValuesOf(Class<E> clazz) {
      E[] enumValues = clazz.getEnumConstants();
      String[] values = new String[enumValues.length];
      for (int i = 0; i < enumValues.length; i++) {
        values[i] = enumValues[i].name();
      }
      return values;
    }

it could be defined on SchemaV3 class

Reply:
you can't use those kind of methods directly in Java annotations (or any kind of dynamic value), but classes are allowed, hence the ValuesProvider class to circumvent this.

not sure why it didn't work for ModelBuilder, processing logic in SchemaMetadata is rather simple:
```java
this.values = annotation.valuesProvider() == ValuesProvider.NULL ? annotation.values() : getValues(annotation.valuesProvider());
...
// If the field is an enum then the values annotation field had better be set. . .
if (is_enum && (null == this.values || 0 == this.values.length)) {
   throw H2O.fail("Didn't find values annotation for enum field: " + this.name);
}
 ```